Automattic/jetpack
# Boost: refactor Critical CSS: Part 1

Refactoring the Critical CSS class without modifying the plugin behavior. 

It's going to take some time to fully refactor Critical CSS, but this is at least a good first step.

Fixes #21725 

### Changes in this pull request:
- The following functionalities are moved to separate classes/files:
    - Critical CSS generation management-related logic.
    - Site path providers.
    - All Critical CSS REST API logic.
    - Critical CSS display in the frontend.
    - Recommendations dismissal and reset.
    - AMP compatibility is moved to a separate file.
- A new base class to handle the options array is added.
- Recommendation admin-ajax calls are converted to REST API calls
- Admin bar compatibility.
- Regeneration admin notices.

#### Jetpack product discussion


pc9hqz-cz-p2

#### Does this pull request change what data or activity we track or use?


No

### Testing instructions:
- Check out this branch - `refactor/boost-critical-css`
- Run `jetpack build plugins/boost`
    - This will install all the dependencies and will build the plugin.
    - (This is needed because a lot of class names/filenames have changed.)
- Run `composer phpcs:lint`
    - Make sure that there are no phpcs errors under `projects/plugins/boost`
- Run all end to end testing as per the instructions in 
    - (Handy commands: `pnpm run env-start && pnpm run tunnel-on` and `pnpm run test-e2e`)
    - Make sure that all the tests pass
- Perform extensive manual testing to make sure the plugin works as expected
